test/www/jxcore/test/thalicryptomanagertests.js

```diff
@@ -0,0 +1,121 @@
+"use strict";
+
+require('./mockmobile');
+var fs = require('fs-extra-promise');
+var path = require('path');
+var crypto = require('crypto');
+var tape = require('wrapping-tape');
+var cryptomanager = require('../thali/thalicryptomanager');
+
+// get the values needed for running the tests
+var configValues = cryptomanager.getConfigValuesForTestingOnly();
+
+var fileLocation = './pkcs12folder';
+
+// test setup & teardown activities
+var test = tape({
+  setup: function(t) {
+    fs.mkdirSync(fileLocation);
+    t.end();
+  },
+  teardown: function(t) {
+    fs.remove(fileLocation, function(err) {
+      if(err) {
+        console.log('could not remove folder - err: ', err);
+      }
+      t.end();
+    });
+  }
+});
+
+test('successfully create a new pkcs12 file and return hash value', 3,
+  function(t) {
+  var errorMessage = null;
+  
+  Mobile.storeGetDocumentsPathReturnArguments(errorMessage, fileLocation);
+  
+  cryptomanager.getPublicKeyHash(function (err, publicKeyHash) {
+    t.equal(err, null);
+
+    var file = path.join(fileLocation, configValues.pkcs12FileName);
+    fs.readFile(file, function (err, pkcs12Content) {
+      t.ifError(err);
+      try {
+        var publicKey = crypto.pkcs12.
+          extractPublicKey(configValues.password, pkcs12Content);
+        if(!publicKey || publicKey.length <= 0) {
+          t.error('extracted public key is invalid');
+          return;
+        }
+      } catch(err) {
+        t.error(err);
```

Comment:
DON'T FIX THIS! But for future reference it would have been cleaner to have wrapped this whole thing in t.doesNotThrow(...., "crypto.pkcs12 failed") rather than use a try/catch.